TMDS_3/include/CGAL/tetrahedron_soup_to_triangulation_3.h

```diff
@@ -0,0 +1,170 @@
+    }
+
+    CGAL::TMDS_3::build_triangulation(tr, points, finite_cells, border_facets);
+
+    CGAL_assertion(CGAL::TMDS_3::internal::is_convex(tr));
+  }
+
+  /** \ingroup PkgTMDS3Functions
+  * builds a 3D triangulation from a soup of tetrahedra.
+  *
+  * @tparam PointRange  a model of the concept `RandomAccessContainer`
+  * whose value type is the point type
+  * @tparam TetrahedronRange a model of the concept `RandomAccessContainer` whose
+  * value type is a model of the concept `RandomAccessContainer` whose value type is `std::size_t`
+  * @tparam Triangulation a 3D triangulation class that has
+  * a vertex base model of `MeshVertexBase_3`
+  * and a cell base model of `MeshCellBase_3`
+  * @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+  *
+  * @param points points of the soup of tetrahedra
+  * @param tets each element in the range describes a tetrahedron using the indices of the points
+  * in `points` (indices 0 to 3), and the associated `Subdomain_index` (index 4). It must
+  * describe a non self-intersecting set of tetrahedra, that cover the convex hull of the
+  * corresponding point set. The tetrahedra must form a valid triangulation with each
+  * pair of neighboring cells sharing exactly one triangle. Combinatorial validity and
+  * validity of the geometric embedding are required.
+  * @param tr the 3D triangulation to be built. If non-empty, `tr` will be cleared prior to
+  * building the triangulation.
+  * @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+  *
+  * \cgalNamedParamsBegin
+  *   \cgalParamNBegin{surface_facets}
+  *     \cgalParamDescription{each element in the range describes a surface facet using the indices of points
+  *       in `points` (indices 0 to 2), and the associated `Surface_patch_index` (index 3)}
+  *     \cgalParamType{a class model of `AssociativeContainer`
+  *                    whose key type is model of `RandomAccessContainer` containing `int`
+  *                    and mapped type is `Tr::Cell::Surface_patch_index`}
+  *     \cgalParamDefault{An empty `boost::unordered_map<std::array<int, 3>, typename Tr::Cell::Surface_patch_index>`}
+  *   \cgalParamNEnd
+  * \cgalNamedParamsEnd
+  *
+  * @pre `points` contains each point only once
+  * @post the output triangulation must be a triangulation of the convex hull of `points`
+  * @post `tr.is_valid()` returns `true`
+  *
+  * @sa `CGAL::Polygon_mesh_processing::polygon_soup_to_polygon_mesh()`
+  */
+  template<typename PointRange,
+           typename TetrahedronRange,
+           typename Triangulation,
+           typename NamedParameters>
+  void tetrahedron_soup_to_triangulation_3(const PointRange& points,
+                                           const TetrahedronRange& tets,
+                                           Triangulation& tr,
+                                           const NamedParameters& np)
```

Comment:
Again: there is an important precondition: the range `tets`, together with `points`, must describe the set of finite tetrahedra of a CGAL triangulation.